src/osd/OSD.cc

```diff
@@ -2086,6 +2145,7 @@ OSD::OSD(CephContext *cct_, ObjectStore *store_,
   osd_op_tp(cct, "OSD::osd_op_tp", "tp_osd_tp",
 	    get_num_op_threads()),
   command_tp(cct, "OSD::command_tp", "tp_osd_cmd",  1),
+  read_tp(cct, "OSD::read_tp", "tp_osd_read",  1),
```

Comment:
Hmm, only one, single `tp_osd_read` thread. This might be a bottleneck under heavy load when many read requests are sent against many different PGs (in other words: `tp_osd_tp` are doing useful work instead of waiting on e.g. `PG::_lock`). Before the changes the reading work is spread across `5` (on HDDs) or `16` (SSDs) `tp_osd_tp` threads.

Please be aware the default value for `osd_pool_default_pg_num` is kept very low (`8`). At teuthology cbt uses `pg_size` between `64` and `256`.